### PR TITLE
List users

### PR DESCRIPTION
**Pull request by francomino-wolox**
## Summary

feat: endpoint get '/users' have an authentication token. Response is paginated.
test: update test '/users GET' with new changes. 

## Known Issues

none

## Trello Card

https://trello.com/c/rkZmGezl

### CODE REVIEW

File: app/controllers/users.ts [hunk 1]
```diff
@@ -8,10 +8,16 @@ import { notFoundError, databaseError, authenticationError } from '../errors';
 import { encode } from '../services/session';
 
 export function getUsers(req: Request, res: Response, next: NextFunction): Promise<Response | void> {
+  const page: number = Number(req.query.page) || 1;
```
Comment (DiegoWolox) [inline app/controllers/users.ts:11]: What would happen if i send a negative value?

File: tests/users.spec.ts [hunk 2]
```diff
@@ -22,13 +22,28 @@ describe('users', () => {
...
+        .post('/users/sessions')
+        .send({ email: user.email, password: noEncryptPass })
         .expect(200)
-        .then((res: request.Response) => {
-          expect(res.body.length).toBe(2);
-          done();
+        .then((res1: request.Response) => {
+          request(app)
+            .get('/users')
+            .set({ Authorization: res1.body.token })
```
Comment (DiegoWolox) [inline tests/users.spec.ts]: you can create a valid token in the code.
no need to call token endpoint.

File: app/controllers/users.ts (hunk 1 above)
Comment (francomino-wolox) [reply app/controllers/users.ts:11]: i added a condition for this case

File: tests/users.spec.ts (hunk 2 above)
Comment (francomino-wolox) [reply tests/users.spec.ts]: created token in the code

**Review by DiegoWolox — APPROVED** [review]

**Pull request merged**